### ISSUE TITLE
NUnit1001 - false positive for arguments with generic parameters

### ISSUE THREAD

**Issue by PiotrKlecha** [nunit/nunit.analyzers#922]
Test values for arguments with generic parameters are incorrectly reported as unassignable, even when they actually match, e.g.:
```cs
[Test]
[TestCase(arg: new[] { 1, 2, 3, })]
[TestCase(arg: new[] { "a", "b", "c", })]
[TestCase(arg: new[] { typeof(int), typeof(string), typeof(double), })]
[TestCase(arg: new[] { 1ul, 2ul, 3ul, })]
public void TestCollectionMethod<T>(IEnumerable<T> collection)
{
	// ... test
}
```
